src/DurableTask.Core/TaskActivityDispatcher.cs

```diff
@@ -182,6 +185,9 @@ async Task OnProcessWorkItemAsync(TaskActivityWorkItem workItem)
                             : null;
                         eventToRespond = new TaskFailedEvent(-1, scheduledEvent.EventId, e.Message, details);
                         this.logHelper.TaskActivityFailure(orchestrationInstance, scheduledEvent.Name, (TaskFailedEvent)eventToRespond, e);
+                        
+                        traceActivity?.SetTag("otel.status_code", "ERROR");
+                        traceActivity?.SetTag("otel.status_description", $"{e.InnerException?.GetType().FullName}: {e.Message}");
```

Comment:
Since this catch block catches anything that derives from `Exception`, I don't think it makes sense to use the inner exception here since we can't make any assumptions about whether there will be an inner exception. Instead, just do this:
```suggestion
                        traceActivity?.SetTag("otel.status_description", $"{e.GetType().FullName}: {e.Message}");
```